cedar-policy-core/src/parser/err.rs

```diff
@@ -803,14 +806,25 @@ impl<T: Into<ParseError>> From<T> for ParseErrors {
 }
 
 impl From<Vec<ParseError>> for ParseErrors {
+    /// Convert a `Vec<ParseError> to a `ParseErrors`, inserting the `Unknown`
+    /// error if necessary.
     fn from(errs: Vec<ParseError>) -> Self {
-        ParseErrors(errs)
+        match errs.split_first() {
+            None => ParseErrors::singleton(ParseError::ToAST(ToASTError::new(
+                ToASTErrorKind::Unknown,
+                Loc::new(0, Arc::from("")),
+            ))),
+            Some((first, rest)) => ParseErrors::new(first.clone(), rest.iter().cloned()),
+        }
     }
 }
 
-impl<T: Into<ParseError>> FromIterator<T> for ParseErrors {
-    fn from_iter<I: IntoIterator<Item = T>>(errs: I) -> Self {
-        ParseErrors(errs.into_iter().map(Into::into).collect())
+impl From<Vec<ToASTError>> for ParseErrors {
+    fn from(errs: Vec<ToASTError>) -> Self {
+        errs.into_iter()
+            .map(Into::into)
+            .collect::<Vec<ParseError>>()
```

Comment:
Can we avoid `.collect()`ing this intermediate `Vec`?  `ParseErrors` does have a `.from_iter()`

Reply:
The `from_iter` function also collects a `Vec` so probably no point. I'm going to leave this comment unaddressed for now since I expect to delete this function in a future PR.